src/woo_search/search_index/api/viewsets/search.py

```diff
@@ -0,0 +1,114 @@
+from django.utils.translation import gettext_lazy as _
+
+from drf_spectacular.utils import extend_schema, extend_schema_view
+from rest_framework import viewsets
+from rest_framework.response import Response
+
+from woo_search.search_index.api.serializers import (
+    SearchResponseSerializer,
+    SearchSerializer,
+)
+
+from ...client import get_client
+from ...constants import SortChoices
+from ...typing import SearchType
+
+
+@extend_schema(tags=["index"])
+@extend_schema_view(
+    create=extend_schema(
+        summary=_("Search"),
+        description=_("Search the publication and document records."),
+        responses={
+            200: SearchResponseSerializer(many=True),
+        },
+    ),
+)
+class SearchViewSet(viewsets.ViewSet):
```

Comment:
I don't see why this needs to be a viewset - a plain `APIView` with `def post` implement is simpler. Using `def create` for what is essentially a read operation is very unnatural. A plain method that is easy to follow from request to response is much nicer:

```py
def post(self, request, *args, **kwargs):
    query_serializer = SearchSerializer(data=request.data)
    query_serializer.is_valid(raise_exception=True)

    ...
```

I'd also opt to use the ES client library DSL: https://elasticsearch-dsl.readthedocs.io/en/latest/search_dsl.html, since it is chainable and you can process it for each bit in the search serializer through if statements if needed.

Reply:
I choose deliberately to not use the DSL for the queries because from what I can read it lacks certain functions that we want to use. A big one would be to boost certain fields.